python/ray/rllib/models/fcnet.py

```diff
@@ -14,35 +14,31 @@ class FullyConnectedNetwork(Model):
     """Generic fully connected network."""
 
     @override(Model)
-    def _build_layers(self, inputs, num_outputs, options):
-        """Process the flattened inputs.
-
-        Note that dict inputs will be flattened into a vector. To define a
-        model that processes the components separately, use _build_layers_v2().
-        """
-
+    def _build_layers_v2(self, input_dict: dict, num_outputs: int, config: dict):
```

Comment:
So there's actually a legacy reason we don't implement _build_layers_v2 for the default FC net -- and that's because we want the network to work for all obs spaces. This only works if the tensor is always the same shape (a flattened vector).

Reply:
I understand. That's a very good reason.